tests/integration/pilot/mcp/testdata/kube/update_service.yaml

```diff
@@ -0,0 +1,37 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tcp-echo-v1
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: tcp-echo
+      version: v1
+  template:
+    metadata:
+      labels:
+        app: tcp-echo
+        version: v1
+    spec:
+      containers:
+      - name: tcp-echo
+        image: docker.io/istio/tcp-echo-server:1.1
```

Comment:
this is better than arbitrary image but should respect the HUB, TAG, and Image Pull Policy. Our tests run hermetically in CI, we load all the images into the cluster directly and don't pull from docker. Is there any reason you need to directly create a Deployment rather than `echo.New`?

Reply:
The only reason I have it in the deployment/service yaml is for the test clarity and redability. Also can I update the service via update_service.yaml if I use echo.New?